Scripts/HW_OOP.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Задание 1 (5 баллов)
+
+# Напишите классы **Chat**, **Message** и **User**. Они должны соответствовать следующим требованиям:
+# 
+# **Chat**:
+# + Должен иметь атрибут `chat_history`, где будут храниться все сообщения (`Message`) в обратном хронологическом порядке (сначала новые, затем старые)
+# + Должен иметь метод `show_last_message`, выводящий на экран информацию о последнем сообщении
+# + Должен иметь метод `get_history_from_time_period`, который принимает два опциональных аргумента (даты с которой и по какую мы ищем сообщения и выдаём их). Метод также должен возвращать объект типа `Chat`
+# + Должен иметь метод `show_chat`, выводящий на экран все сообщения (каждое сообщение в таком же виде как и `show_last_message`, но с разделителем между ними)
+# + Должен иметь метод `recieve`, который будет принимать сообщение и добавлять его в чат
+# 
+# **Message**:
+# + Должен иметь три обязательных атрибута
+#     + `text` - текст сообщения
+#     + `datetime` - дата и время сообщения (встроенный модуль datetime вам в помощь). Важно! Это должна быть не дата создания сообщения, а дата его попадания в чат! 
+#     + `user` - информация о пользователе, который оставил сообщение (какой тип данных использовать здесь, разберётесь сами)
+# + Должен иметь метод `show`, который печатает или возвращает информацию о сообщении с необходимой информацией (дата, время, юзер, текст)
+# + Должен иметь метод `send`, который будет отправлять сообщение в чат
+# 
+# **User**:
+# + Класс с информацией о юзере, наполнение для этого класса придумайте сами
+# 
+# Напишите несколько примеров использования кода, которое показывает взаимодействие между объектами.
+# 
+# В тексте задания намерено не указано, какие аргументы должны принимать методы, пускай вам в этом поможет здравый смысл)
+# 
+# В этом задании не стоит флексить всякими продвинутыми штуками, для этого есть последующие
+# 
+# В этом задании можно использовать только модуль `datetime`
+
+
+import datetime
+
+
+class Chat:
+    def __init__(self, chat_history, chat_history_time_period):
+        self.chat_history = chat_history
+        self.chat_history_time_period = chat_history_time_period
+        
+    def show_last_message(self):
+        self.chat_history[0].show()
+    
+    def get_history_from_time_period(self, first_date, last_date):
+        for message in range(len(self.chat_history)):
+            if (self.chat_history[message].datetime > first_date) and (self.chat_history[message].datetime < last_date):
+                self.chat_history_time_period.insert(0, self.chat_history[message])
+                self.chat_history[message].show()
+                print("\n–––––––––––\n")
+        return self.chat_history_time_period
```

Comment:
1. Не понятно, зачем итерироваться по индексам
2. По итогу у тебя возвращается список сообщений в противоположном порядке
3. `chat_history_time_period` можно было создать как переменную внутри этой функции, ей нет смысла быть атрибутом
4. Метод должен возвращать объект типа `Chat`, но у тебя он возвращает список
![image](https://user-images.githubusercontent.com/48160953/226216719-7634254a-58ff-47ae-817a-8623ef73ec99.png)

```suggestion
        chat_history_time_period = []
        for message in self.chat_history:
            if first_date <= self.chat_history[message].datetime <= last_date: 
                self.chat_history_time_period.append(message)
        return Chat(chat_history_time_period)
```